src/process.c

```diff
@@ -463,59 +466,450 @@ cc_run_hook(struct oci_cfg_hook* hook, const gchar* state,
 	return result;
 }
 
+
+/*!
+ * Obtain the network configuration by querying the network namespace.
+ *
+ * \param[in,out] config \ref cc_oci_config.
+ *
+ * \return \c true on success, else \c false.
+ */
+static gboolean
+cc_oci_vm_netcfg_get (struct cc_oci_config *config)
+{
+	if ((! config)) {
+		return false;
+	}
+
+	/* TODO: We need to support multiple networks */
+	if ( !cc_oci_network_discover ("eth0", config)) {
+		g_critical("Network discovery failed");
+		return false;
+	}
+
+	return true;
+}
+
+/*!
+ * Obtain the network configuration by querying the network namespace,
+ * returning the data in "name=value" format in a
+ * dynamically-allocated string vector.
+ *
+ * \param config \ref cc_oci_config.
+ *
+ * \return \c string vector on success, , else \c NULL.
+ */
+static gchar **
+cc_oci_vm_netcfg_to_strv (struct cc_oci_config *config)
+{
+	gchar  **cfg = NULL;
+	gsize    count = 0;
+
+	if (! config) {
+		return NULL;
+	}
+
+	if (! cc_oci_vm_netcfg_get (config)) {
+		return NULL;
+	}
+
+	if (config->net.hostname)    count++;
+	if (config->net.gateway)     count++;
+	if (config->net.dns_ip1)     count++;
+	if (config->net.dns_ip2)     count++;
+	if (config->net.mac_address) count++;
+	if (config->net.ipv6_address)  count++;
+	if (config->net.ip_address)  count++;
+	if (config->net.subnet_mask) count++;
+	if (config->net.ifname)      count++;
+	if (config->net.bridge)      count++;
+	if (config->net.tap_device)  count++;
```

Comment:
I know that sometimes does not make sense to add braces when we have only one instruction inside a control statement (if, else, ....), missing braces can cause logic errors or even security vulnerabilities, so please add braces ;)